Eevee.Sleep.Bot/Modules/SlashCommands/RoleManagementSlashModule.cs

```diff
@@ -0,0 +1,131 @@
+using Discord;
+using Discord.Interactions;
+using JetBrains.Annotations;
+using Eevee.Sleep.Bot.Controllers.Mongo;
+using Eevee.Sleep.Bot.Utils;
+using Eevee.Sleep.Bot.Enums;
+using Discord.WebSocket;
+
+namespace Eevee.Sleep.Bot.Modules.SlashCommands;
+
+[Group("role", "Commands for managing the roles on a user.")]
+public class RoleManagementSlashModule : InteractionModuleBase<SocketInteractionContext> {
+    [SlashCommand("display", "Select which of the role you have to display.")]
+    [UsedImplicitly]
+    public async Task DisplayRoleAsync() {
+        await Context.Interaction.RespondAsync(
+            "Select your role to display.",
+            components: DiscordMessageMaker.MakeRoleSelectButton(
+                ownRoles: DiscordRoleContoller.FindRoleIdsByUserId(Context.User.Id), 
+                trackedRoles: DiscordRoleContoller.FindAllTrackedRoles(),
+                buttonId: ButtonId.RoleChanger
+            ),
+            ephemeral: true
+        );
+
+        // Ephemeral messages cannot be deleted by the button handler, so the message containing the button should be deleted after 30 seconds.
+        await Task.Delay(TimeSpan.FromSeconds(30));
+        await Context.Interaction.DeleteOriginalResponseAsync();
+    }
+
+    [SlashCommand("add", "Add the role ownership of a user.")]
+    [UsedImplicitly]
+    public async Task AddRoleAsync() {
+        var user = Context.User is SocketGuildUser matchedUser ? matchedUser : null;
+        await Context.Interaction.RespondAsync(
+            "Select your role to add.",
+            components: DiscordMessageMaker.MakeRoleSelectButton(
+                // Find the role ids that the user does not have
+                ownRoles: DiscordRoleContoller
+                    .FindRoleIdsByUserId(Context.User.Id)
+                    .Except(user?.Roles.Select(x => x.Id) ?? [])
+                    .ToArray(), 
+                trackedRoles: DiscordRoleContoller.FindAllTrackedRoles(),
+                buttonId: ButtonId.RoleAdder
+            ),
+            ephemeral: true
+        );
+
+        // Ephemeral messages cannot be deleted by the button handler, so the message containing the button should be deleted after 30 seconds.
+        await Task.Delay(TimeSpan.FromSeconds(30));
+        await Context.Interaction.DeleteOriginalResponseAsync();
+    }
+
+    [SlashCommand("remove", "Remove the role ownership of a user.")]
+    [UsedImplicitly]
+    public async Task RemoveRoleAsync() {
+        var user = Context.User is SocketGuildUser matchedUser ? matchedUser : null;
```

Comment:
If the `user` is null (not a `SocketGuildUser`), should we even handle? Feels like we can just respond with an error-like message and stop.